### PR TITLE
[Mono] Add osx aot offsets & generate aot compiler packs for x64 and arm64

### PR DESCRIPTION
**Pull request by steveisok**
Since Xam.mac can toggle between runtimes, it may make sense to provide an AOT option when using mono.

### CODE REVIEW

File: src/mono/CMakeLists.txt [hunk 1]
```diff
@@ -119,7 +119,11 @@ if(NOT AOT_TARGET_TRIPLE STREQUAL "")
     set(MONO_OFFSETS_FILE "${AOT_OFFSETS_FILE}")
   endif()
   if(AOT_TARGET_TRIPLE STREQUAL "x86_64-apple-darwin10")
-    set(TARGET_SYSTEM_NAME "iOS")
+    if(REAL_TARGET_OS STREQUAL "osx")
```
Comment (lambdageek) [inline src/mono/CMakeLists.txt:122]: I'm surprised this is necessary.

Actually I'm not sure why we have 3 sections -  one that checks `AOT_TARGET_TRIPLE`, and sets `TARGET_` properties, one that checks `CMAKE_SYSTEM_NAME` and sets the `HOST_` properties,  and a third one that checks `TARGET_SYSTEM_NAME` and sets `TARGET_` properties.

Don't `AOT_TARGET_TRIPLE` and `TARGET_SYSTEM_NAME` checks set the same variables? Does that mean the `TARGET_SYSTEM_NAME` settings are overwriting whatever `AOT_TARGET_TRIPLE` sets up?  Why do we need both?

Comment (steveisok) [reply src/mono/CMakeLists.txt:122]: Not sure if we need it.  I was trying to fit within what was there and it felt weird to set the `TARGET_SYSTEM_NAME` to iOS when I know it should be Darwin

Comment (lambdageek) [reply src/mono/CMakeLists.txt:122]: Oh, I am misreading what is happening.  We set AOT_TARGET_TRIPLE from the outside.  And then that sets TARGET_SYSTEM_NAME and then TARGET_SYSTEM_NAME is used to set other target properties.  So it's not redundant.  The triple is the source of truth.

It's unfortunate that we use the same triple for ios and mac.   Did we do that in mono/mono, too? @vargaz ?

Comment (vargaz) [reply src/mono/CMakeLists.txt:122]: We has no cross compiler to osx previously, probably just used the normal host mono. If we need it, it should use a separate triple.

Comment (steveisok) [reply src/mono/CMakeLists.txt:122]: Is there anything you would suggest?

Comment (vargaz) [reply src/mono/CMakeLists.txt:122]: Something like x86_64-apple-macos

Comment (akoeplinger) [reply src/mono/CMakeLists.txt:122]: according to https://reviews.llvm.org/D40998 they use `x86_64-apple-macosx` (note the `x` at the end) for clang's `-target` but if I understand correctly we're not passing the triple down to clang here so this is just for us?

**Pull request closed without merging**